src/components/Firestore/DocumentPreview/FieldPreview.tsx

```diff
@@ -54,17 +54,19 @@ const FieldPreview: React.FC<{
   let childFields = null;
   if (isMap(state)) {
     // Inline editor for new field will default to key: ''
-    childFields = Object.keys(state).map(childLeaf => {
-      const childPath = [...path, childLeaf];
-      return (
-        <FieldPreview
-          key={childLeaf}
-          path={childPath}
-          documentRef={documentRef}
-          maxSummaryLen={maxSummaryLen}
-        />
-      );
-    });
+    childFields = Object.keys(state)
+      .sort((fieldA, fieldB) => fieldA.localeCompare(fieldB))
```

Comment:
Would you mind extracting a function like `compareFirestoreKeys` for all uses below? That function can live in `util.ts` or somewhere close. We may need to change the comparison or apply polyfills globally.

(Side note: Firestore uses a different sorting method than `localeCompare`, but I'd still suggest we merge this for now since it is close enough for most browsers in English locales and we can improve it later.)

Reply:
Ok! Do you want to use that function just in that place, or use it also for maps inside firestore documents?

Reply:
Please use the function for all new sorting you've introduced, thanks! If you notice any other places where we're sorting keys, feel free to refactor those too.

As far as string comparison goes, Firestore production uses UTF-8 encoded byte order ([docs](https://firebase.google.com/docs/firestore/manage-data/data-types)). Unfortunately, it is very hard to get that right in client-side JavaScript (even if you use [hacks](http://ecmanaut.blogspot.com/2006/07/encoding-decoding-utf8-in-javascript.html)), so please don't stress yourself to get that in this PR. A easily swappable function plus useful comments that explains the situation is good enough.

Reply:
Ok, I will not waste time on that since it seems pretty tricky. THanks for the info!